snappy/snapd_client.go

```diff
@@ -82,3 +94,93 @@ func (a *ClientAdapter) Remove(name string, options *client.SnapOptions) (string
 func (a *ClientAdapter) ServerVersion() (*client.ServerVersion, error) {
 	return a.snapdClient.ServerVersion()
 }
+
+// internal
+func readNTPServer() string {
+	timesyncd, err := ini.Load(timesyncdPath)
+	if err != nil {
+		log.Println("readNTPServer: unable to read /etc/system/timesyncd.conf")
+		return ""
+	}
+
+	section, err := timesyncd.GetSection("Time")
+	if err != nil || !section.HasKey("NTP") {
+		log.Println("readNTPServer: no NTP servers are set")
+		return ""
+	}
+
+	return section.Key("NTP").Strings(" ")[0]
+}
+
+// Write back directly, rather than write-tmp(in same dir)+rename, as per the
+// internals of ini package, although better, write accessible locations may
+// not be on the same filesystem, removing the advantage.
+//  : timeserver-control grants rw access to /etc/systemd/timesyncd.conf
+func saveTimeSyncd(confFile *ini.File) error {
+	timesyncConf, err := os.OpenFile(timesyncdPath, os.O_WRONLY|os.O_TRUNC, 0644)
```

Comment:
(nitpick) It looks like `os.Create(timesyncdPath)` could be used here which is slightly shoter.